src/env/mod.rs

```diff
@@ -35,7 +38,72 @@ pub trait Env {
     // /// Get a reference to stdin
     // fn stdin(&self) -> impl io::Read;
 
+    /// Make relative paths absolute.
     fn in_cwd<'a>(&self, path: &'a impl AsRef<Path>) -> Cow<'a, Path>;
+
+    /// Create and open a file.
+    fn fs_create_new(&self, path: impl AsRef<Path>) -> Result<File> {
+        let path = path.as_ref();
+        let abs_path = self.in_cwd(&path);
+        File::create_new(abs_path)
+            .map_err(|err| format!("cannot create '{}': {err}", path.display()).into())
+    }
+
+    /// Rename a path.
+    fn fs_rename(&self, old: impl AsRef<Path>, new: impl AsRef<Path>) -> Result<()> {
+        let (old, new) = (old.as_ref(), new.as_ref());
+        let abs_old = self.in_cwd(&old);
+        let abs_new = self.in_cwd(&new);
+        std::fs::rename(abs_old, abs_new).map_err(|err| {
+            format!(
+                "could not move '{}' to '{}': {err}",
+                old.display(),
+                new.display()
+            )
+            .into()
+        })
+    }
+
+    /// Create a symlink.
+    #[cfg(unix)]
+    fn fs_symlink(&self, target: impl AsRef<Path>, link: impl AsRef<Path>) -> Result<()> {
+        let (target, link) = (target.as_ref(), link.as_ref());
+        let target_path = self.in_cwd(&target);
+        let link_path = self.in_cwd(&link);
+        std::os::unix::fs::symlink(target_path, link_path).map_err(|err| {
+            format!(
+                "could not create symlink '{}' to '{}': {err}",
+                link.display(),
+                target.display(),
+            )
+            .into()
+        })
+    }
+
+    /// Create a symlink, overwriting if it already exists.
+    #[cfg(unix)]
+    fn fs_symlink_force(&self, target: impl AsRef<Path>, link: impl AsRef<Path>) -> Result<()> {
```

Comment:
I'm not sure about adding higher-level API's than `std` here.

Reply:
For the purpose of CLI programs, `std` actually provides relatively little.  In most languages, CLI codebases tend to have their own utility modules with these kinds of specialized filesystem / OS operations, often matching up with common command-line tools (e.g. `symlink_force` is basically `ln -sf`).  I think we're going to accumulate these kinds of utility functions _somewhere_, and while `Env` isn't necessarily the right place, I think it's certainly the most convenient.

Reply:
I'm confused, you said yourself that adding a lot of functions to `Env` is probably not the right way to go? Besides, it depends on the function, I think we should keep them with the tool as long as they're not used in other tools.

This kind of muddles the idea of `Env` into both low-level environment interaction a `util` module. A function that takes `Env` as first argument is just as effective.

Don't consider this blocking though, it's alright.

Reply:
I suspect symlinks will show up in other commands as well.  I'll make a `util` module that provides these functions instead.